src/Host/UserProfile/Impl/RUserProfileService.cs

```diff
@@ -36,25 +39,52 @@ partial class RUserProfileService : ServiceBase {
                 SecurityIdentifier sid = new SecurityIdentifier(WellKnownSidType.WorldSid, null);
                 PipeAccessRule par = new PipeAccessRule(sid, PipeAccessRights.ReadWrite, System.Security.AccessControl.AccessControlType.Allow);
                 ps.AddAccessRule(par);
-                using (NamedPipeServerStream server = new NamedPipeServerStream("RUserCreatorPipe", PipeDirection.InOut, 1, PipeTransmissionMode.Byte, PipeOptions.Asynchronous, 1024, 1024, ps))
-                using (BinaryReader reader = new BinaryReader(server))
-                using (BinaryWriter writer = new BinaryWriter(server)) {
+                using (NamedPipeServerStream server = new NamedPipeServerStream("Microsoft.R.Host.UserProfile.Creator{b101cc2d-156e-472e-8d98-b9d999a93c7a}", PipeDirection.InOut, 1, PipeTransmissionMode.Byte, PipeOptions.Asynchronous, 1024, 1024, ps)) {
                     try {
                         await server.WaitForConnectionAsync(ct);
 
-                        string username = reader.ReadString();
-                        string domain = reader.ReadString();
-                        string password = reader.ReadString();
+                        byte[] request = new byte[1024];
+                        int bytesRead = 0;
 
-                        var result = RUserProfileCreator.Create(username, domain, password);
+                        while(bytesRead == 0 && !ct.IsCancellationRequested) {
+                            bytesRead = await server.ReadAsync(request, 0, request.Length, ct);
+                        }
+                        
+                        byte[] data = new byte[bytesRead];
+                        Array.Copy(request, data, bytesRead);
 
-                        writer.Write(result.Win32Result);
-                        writer.Write(result.ProfileExists);
-                        writer.Write(result.ProfilePath);
+                        string json = Encoding.Unicode.GetString(data);
+                        JArray dataArray = JArray.Parse(json);
```

Comment:
This can throw if incoming JSON is invalid - needs to be handled alongside `IOException` below.